ob1k-consul/src/main/java/com/outbrain/ob1k/consul/LoadBalancer.java

```diff
@@ -0,0 +1,10 @@
+package com.outbrain.ob1k.consul;
+
+import java.util.List;
+
+public interface LoadBalancer {
+
+  List<String> provideTargets(final List<String> targets, final int targetsNum);
+
+  default void onTargetsChanged(final List<String> newTargets) {}
```

Comment:
You should probably move the targets to the implementations, and remove them from the targets provider and from the provideTargets() args